### PR TITLE
Add nvimagecodec libs to DALI_EXCLUDES + test utils to dump mismatched images

### PR DESCRIPTION
**Pull request by jantonguirao**
## Category:

**Bug fix**

## Description:

Fixes build with WITH_DYNAMIC_NVIMGCODEC=OFF

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/test_utils.py [hunk 1]
```diff
@@ -186,6 +186,33 @@ def get_absdiff(left, right):
         _check_absdiff()
     return _get_absdiff(left, right)
 
+def dump_as_core_artifacts(image_info, lhs, rhs, iter=None, sample_idx=None):
+    import_numpy()
+    import_pil()
+    from pathlib import Path
+    path = "/opt/dali" if os.path.exists("/opt/dali") else os.getcwd()
```
Comment (mzient) [inline dali/test/python/test_utils.py:193]: That's not good. A regular user who happens to have DALI there won't be able to write to `/opt`.

Comment (jantonguirao) [reply dali/test/python/test_utils.py]: added a check for write permissions

**Review by mzient — APPROVED** [review]

File: dali/operators/imgcodec/image_decoder.h [hunk 2]
```diff
@@ -79,6 +79,8 @@ static constexpr size_t kHostAlignment = 64;  // cache alignment
 inline int static_dali_device_malloc(void *ctx, void **ptr, size_t size, cudaStream_t stream) {
   auto *mr = static_cast<mm::device_async_resource *>(ctx);
   try {
+    std::cout << "device_malloc this_thread[" << std::this_thread::get_id() << "] stream=" << stream
+              << "\n";
```
Comment (klecki) [inline dali/operators/imgcodec/image_decoder.h:83]: Is this intended or it snuck from some debugging code?

**Review by klecki — APPROVED** [review]

File: dali/operators/imgcodec/image_decoder.h (hunk 2 above)
Comment (jantonguirao) [reply dali/operators/imgcodec/image_decoder.h]: removed it

**Comment by jantonguirao** [comment]
!build

**Pull request merged**